v8.4.0 (#17948)
mui/mui-x: CHANGELOG.md

- 🔺 Support regular [`pyramid` variation in the `<FunnelChart />` component](https://mui.com/x/react-charts/funnel/#pyramid-chart):

  <img width="398" alt="Pyramid funnel chart" src="https://github.com/user-attachments/assets/90ccb221-3a48-4ffa-8878-89c6db16da86" />

- 📚 Documentation improvements
- 🌎 Improve Icelandic (is-IS) locale on the Data Grid
- 🐞 Bugfixes

Special thanks go out to the community members for their valuable contributions:
@aizerin, @arminmeh, @campmarc, @jyash97, @mapache-salvaje, @noraleonte, @nusr, @ragnarr18, @romgrk, @whereisrmsqhs.
Following are all team members who have contributed to this release:
@alexfauquette, @bernardobelchior, @flaviendelangle, @JCQuintas, @KenanYusuf, @LukasTy, @MBilalShafi, @rita-codes.

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

### Data Grid

#### `@mui/x-data-grid@8.4.0`

- [DataGrid] Fix content rendering for large rows while using automatic page size (#14737) @campmarc
- [DataGrid] Fix disabled typography variants crashing grid (#17934) @KenanYusuf
- [DataGrid] Fix tree data demo crash (#17904) @MBilalShafi
- [DataGrid] Use `exclude` selection model type if quick filter does not have actual values (#17899) @arminmeh

> - [DataGrid] Fix copy a selection of cells (#16797) @nusr

Reviewer: ```suggestion
- [DataGrid] Fix clipboard copy behavior for cell ranges with empty cells (#16797) @nusr
```

- [l10n] Improve Icelandic (is-IS) locale (#17915) @ragnarr18

#### `@mui/x-data-grid-pro@8.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@8.4.0`, plus:

> - [DataGridPro] Add `aria-expanded` attribute to the Master detail toggle button (#17122) @whereisrmsqhs

Reviewer: ```suggestion
- [DataGridPro] Add `aria-expanded` attribute to the master detail toggle button (#17122) @whereisrmsqhs
```

> - [DataGridPro] Prevent row state loss with server side lazy loading (#17743) @arminmeh

Reviewer: ```suggestion
- [DataGridPro] Preserve row state during server-side lazy loading (#17743) @arminmeh
```

- [DataGridPro] Prevent text selection when reordering rows (#16568) @jyash97

#### `@mui/x-data-grid-premium@8.4.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.4.0`.

### Date and Time Pickers

#### `@mui/x-date-pickers@8.4.0`

- [fields] Ensure fresh `disabled` value is used when focusing or clicking (#17914) @aizerin
- [fields] Improve the field controlled edition (#17816) @flaviendelangle
- [pickers] Fix `PickersTextField` overflow (#17942) @noraleonte

#### `@mui/x-date-pickers-pro@8.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-date-pickers@8.4.0`.

### Charts

#### `@mui/x-charts@8.4.0`

- [charts] Add grouped axes demo (#17848) @bernardobelchior

> - [charts] Add zoom slider tooltip (#17733) @bernardobelchior
> - [charts] Clean and document Heatmap Tooltip (#17933) @alexfauquette
> - [charts] Enable tooltip disable portal (#17871) @alexfauquette
> - [charts] Improve performance in scatter chart (#17849) @bernardobelchior
> - [charts] Introduce `Pyramid` chart (#17783) @JCQuintas

Reviewer 2: ```suggestion
- [charts] Enable tooltip disable portal (#17871) @alexfauquette
- [charts] Improve performance in scatter chart (#17849) @bernardobelchior
```

Reviewer 3: @michelengelen In line with the labels discussion, should we have the `[tag]` be automatically added to PR's based on the actual labels? 😆

- [charts] Recreate `isPointInside` less often (#17850) @bernardobelchior
- [charts] Try fix for flaky `useAnimate` test (#17777) @JCQuintas
- [charts] Add `isXInside` and `isYInside` (#17911) @bernardobelchior

#### `@mui/x-charts-pro@8.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.4.0`, plus:

- [charts-pro] Add size for zoom slider (#17736) @bernardobelchior
- [charts-pro] Update zoom slider nomenclature (#17938) @bernardobelchior
- [charts-pro] Fix error when importing rasterizehtml (#17897) @bernardobelchior

### Tree View

#### `@mui/x-tree-view@8.4.0`

- [TreeView] Add `getItemChildren` prop in `RichTreeView` (#17894) @rita-codes
- [TreeView] Add a method in the `apiRef` to toggle the editing status of an item (#17768) @rita-codes
- [TreeView] Add missing sx prop on the Tree Item component (#17930) @flaviendelangle

#### `@mui/x-tree-view-pro@8.4.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-tree-view@8.4.0`.

### Docs

> - [docs] Add introductory text to Data Grid component pages (#17902) @KenanYusuf

Reviewer: ```suggestion
- [docs] Add a recipe for drag and drop row grouping (#17638) @KenanYusuf
- [docs] Add introductory text to Data Grid component pages (#17902) @KenanYusuf
```

> - [docs] Refactor embedded Codesandbox - DataGrid Quickstart page (#17749) @rita-codes

Reviewer: ```suggestion
- [docs] Refactor embedded CodeSandbox on Data Grid—Quickstart page (#17749) @rita-codes
```

> - [docs] Remove double border on DataGrid- Quickstart (#17932) @rita-codes

Reviewer: ```suggestion
- [docs] Remove double border on Data Grid—Quickstart demo (#17932) @rita-codes
```
